### PR TITLE
Cargo framework documentation

### PR DESCRIPTION
**Pull request by alganthe**

### CODE REVIEW

File: wiki/framework/cargo-framework.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+</div>
+
+
+## 2. Events
+
+### 2.1 Listenable
+
+Event Name | Description | Passed Parameter(s) | Locality
+---------- | ----------- | ------------------- | --------
+`cargoLoaded` | Loaded cargo. | ` [_item, _vehicle]` | Global
```
Comment (jonpas) [inline wiki/framework/cargo-framework.md]: Event descriptions are usually without a final dot when they are that short (not really a description but there is no need for longer).

Also one too much space in passed parameters on that line.

File: wiki/framework/cargo-framework.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        ace_cargo_size = 4;  // The cargo space the object takes
+        ace_cargo_canLoad = 1;  // Enables the object to be loaded (1 = yes, 0 = no, default = 1)
+
+    };
+};
+```
+
+<div class="panel callout">
+    <h5>Note:</h5>
+    <p>ace_cargo_hasCargo and ace_cargo_canLoad are only needed if you aren't inheriting from any of BI base classes or if you are trying to disable loading for a specific vehicle / object</p>
```
Comment (jonpas) [inline wiki/framework/cargo-framework.md]: Final dot missing.

File: wiki/framework/cargo-framework.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+```
+
+### 1.2 Making an object loadable
+
+```c++
+class staticBananaLauncher {
+    class yourVehicleBaseClass {
+        ace_cargo_size = 4;  // The cargo space the object takes
+        ace_cargo_canLoad = 1;  // Enables the object to be loaded (1 = yes, 0 = no, default = 1)
+
```
Comment (jonpas) [inline wiki/framework/cargo-framework.md]: Unneeded space.

File: wiki/framework/cargo-framework.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+## 1. Config Values
+
+### 1.1 Preparing a vehicle for ACE3 cargo
+
+```c++
+class CfgVehicles {
+    class yourVehicleBaseClass {
+        ace_cargo_space = 4;  // The cargo space your vehicle has
+        ace_cargo_hasCargo = 1;  // Enables cargo to be loaded inside the vehicle (1 = yes, 0 = no, default = 1)
+
```
Comment (jonpas) [inline wiki/framework/cargo-framework.md]: Unneeded space.

File: wiki/framework/cargo-framework.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+---
+
+## 1. Config Values
+
+### 1.1 Preparing a vehicle for ACE3 cargo
+
+```c++
+class CfgVehicles {
+    class yourVehicleBaseClass {
+        ace_cargo_space = 4;  // The cargo space your vehicle has
```
Comment (jonpas) [inline wiki/framework/cargo-framework.md]: Unnecessary `The`, let's try to keep it short and simple.

**Comment by PabstMirror** [comment]
Fix Issue #2741

Maybe add the following that was recently asked:
```
## 3. Ojbect Variables
To disable cargo for a mission object use:
\```c++
this setVariable ["ace_cargo_size", -1];
\```
```

File: wiki/framework/cargo-framework.md [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+        ace_cargo_hasCargo = 1;  // Enables cargo to be loaded inside the vehicle (1 = yes, 0 = no, default = 1)
```
Comment (jonpas) [inline wiki/framework/cargo-framework.md]: Everywhere else the style used is: `(Optional) Text (0-no, 1-yes)`

Here I'd just remove the optional/default note though, if a developer doesn't inherit from a class that has this config entry set then that won't be true. It's more of a general inheritance guide then, no need to be here.

File: wiki/framework/cargo-framework.md [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+};
+```
+
+### 1.2 Making an object loadable
+
+```c++
+class staticBananaLauncher {
+    class yourVehicleBaseClass {
+        ace_cargo_size = 4;  // The cargo space the object takes
+        ace_cargo_canLoad = 1;  // Enables the object to be loaded (1 = yes, 0 = no, default = 1)
```
Comment (jonpas) [inline wiki/framework/cargo-framework.md]: Same as https://github.com/acemod/ACE3/pull/2775/files#r42531594

**Comment by alganthe** [comment]
I'll add that to the Mission maker section, the framework entries are mostly for addon makers

**Comment by Glowbal** [comment]
I think that information is also relevant for addon makers.

**Comment by alganthe** [comment]
I'll add the needed information.

**Comment by alganthe** [comment]
Here we go @jonpas, I also added the part about disabling cargo for an object mentioned by @PabstMirror.

**Pull request merged**